src/utils/promise.js

```diff
@@ -62,6 +62,21 @@ export function some(promises, count = 1) {
  * Returns a Deferred struct, which holds a pending promise and its associated
  * resolve and reject functions.
  *
+ * This is preferred instead of creating a Promise instance to extract the
+ * resolve/reject functions yourself:
+ *
+ * ```
+ * // Avoid doing
+ * let resolve;
+ * const promise = new Promise(res => {
+ *   resolve = res;
+ * });
+ *
+ * // Good
+ * const deferred = new Deferred();
+ * const { promise, resolve } = deferred;
+ * ```
+ *
  * @template T
  */
 export class Deferred {
```

Comment:
I suggest moving this class to the top of the file, since more people will want to use `Deferred` than `some`.